backend/danswer/chat/chat_utils.py

```diff
@@ -508,3 +511,28 @@ def extract_citations_from_stream(
             yield DanswerAnswerPiece(answer_piece="[" + curr_segment)
         else:
             yield DanswerAnswerPiece(answer_piece=curr_segment)
+
+
+_MAX_MISC_TOKENS = 2000
+
+
+def get_max_misc_tokens_for_chat() -> int:
+    """Fetches the maximum allows size of the "non context document" part of the
+    prompt fed to the LLM to generate the final response."""
+
+    # right now, this is a fixed value regardless of the llm context size
+    # in the future, this should be a function of the context size
+    return _MAX_MISC_TOKENS
+
+
+def compute_max_document_tokens(persona: Persona | None) -> int:
```

Comment:
Actually so the logic previously was:
1. Create the whole set of messages as if there is no limit on the LLM context.
2. See how much over the limit we are in actuality
3. Drop the number of messages necessary from the beginning of the history

So there isn't a set number of tokens reserved for the documents, the documents can effectively be almost as large as the entire input limit

Reply:
I think we may need a limit on the user input length, what do you think? Backend can throw an error if it's too long and frontend can display some error? Can just be some global setting for how long a max user input is?

Reply:
Final implementation:
1. Amount tokens available for docs is: model context limit - output tokens (1k) - system prompt - task prompt template - expected user input "limit" => all this is used to calculate the max tokens for the doc selection UI step
2. User provides actual input => same as above but use the actual user input
- if user input is shorter, we can use less truncation
- if user input is longer, use more truncation, just hide it from the end user, they don't need to know shhhhh